allennlp/predictors/predictor.py

```diff
@@ -109,7 +109,11 @@ def get_gradients(self, instances: List[Instance]) -> Tuple[Dict[str, Any], Dict
             )
 
             loss = outputs["loss"]
-            self._model.zero_grad()
+            # Zero gradients.
+            # NOTE: this is actually more efficient than calling `self._model.zero_grad()`
+            # because it avoids a read op when the gradients are first updated below.
+            for p in self._model.parameters():
```

Comment:
Here, the parameters come from the model. In other places they come from the optimizer. Does that matter?

Reply:
I don't think it matters much. The only difference here is that we may be setting `p.grad` to `None` for some parameters where `p.grad` is already `None`.